protocol-designer/src/steplist/fieldLevel/processing.js

```diff
@@ -6,19 +6,19 @@ export type ValueCaster = (value: mixed) => mixed
  **  Value Casters   **
  **********************/
 
-// TODO: account for floats and negative numbers
+// Mask to number now allows for 0 and negative numbers, for decimals use maskToFloat
 export const maskToNumber = (rawValue: mixed): mixed => {
-  if (!rawValue) return null
-  let castValue = Number(rawValue)
-  if (Number.isNaN(castValue)) {
-    const cleanValue = String(rawValue).replace(/[\D]+/g, '')
-    return cleanValue
-  } else {
-    return castValue
-  }
+  if (!rawValue) return Number(rawValue)
+  const rawNumericValue =
+    typeof rawValue === 'string'
+      ? rawValue.replace(/[^-0-9]/, '')
```

Comment:
As written, this will only strip out the first occurrence of a non-numerical thing in the string. I don't think this was the intention, so you probably want to add the global flag to this regex

```suggestion
      ? rawValue.replace(/[^-0-9]/g, '')
```

Reply:
Updated! Good catch.